test/testdata/structured-config-refs/go.mod

```diff
@@ -0,0 +1,90 @@
+module structured-config-refs
```

Comment:
I guess I was a bit biased here in fact 😅, I wrote the example/test program in Go because I wanted to be sure that we could deserialize a structured, complex config in the way that we usually do in Go